AttributeError: 'CustomCommandExecutor' object has no attribute 'step_settings'

Using this test:

```python
from addons.web_extensions import WebExtensions
from selenium.webdriver.common.by import By
from src.testproject.classes import DriverStepSettings, StepSettings
from src.testproject.decorator import report_assertion_errors
from src.testproject.sdk.drivers import webdriver
from src.testproject.sdk.drivers.actions import Actions
import pytest



@pytest.fixture()
def driver():
    driver = webdriver.Chrome(token="",
                              project_name="Test",
                              job_name="Test")
    step_settings = StepSettings(timeout=2000)
    with DriverStepSettings(driver, step_settings):
        yield driver
    driver.quit()


@report_assertion_errors
def test_main(driver):
    # Test Parameters
    ApplicationURL = "https://mobii.ai"  # Auto generated application URL parameter

    # 1. Navigate to '{ApplicationURL}'
    # Navigates the specified URL (Auto-generated)
    driver.get(f'{ApplicationURL}')

    # 2. Click 'IMG4'
    # step_settings = StepSettings(always_pass=True)
    # with DriverStepSettings(driver, step_settings):
    #     img4 = driver.find_element(By.XPATH,
    #                                "//swiper/div/button/img")
    #     img4.click()

    # 3. Set window size to ('760','1020')
    driver.addons().execute(
        WebExtensions.setwindowsize(
            width="760",
            height="1020"))

    # 4. Pause for '1000'ms
    driver.pause(milliseconds=1000)

    # 5. Click 'IMG31'
    img31 = driver.find_element(By.XPATH,
                                "//div[1]/div/div[5]/a//img")
    img31.click()

    # 6. Is 'IMG32' visible?
    img32 = driver.find_element(By.XPATH,
                                "//div/div/a/img")
    assert img32.is_displayed()

    # 7. Click 'IMG9'
    img9 = driver.find_element(By.XPATH,
                               "//a[5]/img[1]")
    img9.click()

```

Results in:

`FAILED test_1_244_demo.py::test_main - AttributeError: 'CustomCommandExecutor' object has no attribute 'step_settings'`

At step:

```
    # 4. Pause for '1000'ms
    driver.pause(milliseconds=1000)
```

Using SDK 0.65.2

--- a/src/testproject/sdk/internal/helpers/reporting_command_executor.py
+++ b/src/testproject/sdk/internal/helpers/reporting_command_executor.py
@@ -285,24 +285,24 @@ class ReportingCommandExecutor:
         """
 
         # Handling sleep before execution
-        self.step_helper.handle_sleep(sleep_timing_type=self.step_settings.sleep_timing_type,
-                                      sleep_time=self.step_settings.sleep_time)
+        self.step_helper.handle_sleep(sleep_timing_type=self.settings.sleep_timing_type,
+                                      sleep_time=self.settings.sleep_time)
         # Sleep for...
         time.sleep(milliseconds / 1000.0)
 
         # Handling sleep after execution
-        self.step_helper.handle_sleep(sleep_timing_type=self.step_settings.sleep_timing_type,
-                                      sleep_time=self.step_settings.sleep_time,
+        self.step_helper.handle_sleep(sleep_timing_type=self.settings.sleep_timing_type,
+                                      sleep_time=self.settings.sleep_time,
                                       step_executed=True)
         result, step_message = self.step_helper.handle_step_result(
             step_result=True,
-            invert_result=self.step_settings.invert_result,
-            always_pass=self.step_settings.always_pass)
+            invert_result=self.settings.invert_result,
+            always_pass=self.settings.always_pass)
 
         # Handle screenshot condition
-        screenshot = self.step_helper.take_screenshot(self.step_settings.screenshot_condition, result)
         self.report().step(description=f'Pause for {{{{{milliseconds}}}}} ms',
                            message=step_message,
                            inputs={"milliseconds": milliseconds},
                            passed=result,
                            screenshot=screenshot)
+        screenshot = self.step_helper.take_screenshot(self.settings.screenshot_condition, result)

--- a/src/testproject/sdk/internal/helpers/reporting_command_executor.py
+++ b/src/testproject/sdk/internal/helpers/reporting_command_executor.py
@@ -24,6 +24,7 @@ from src.testproject.helpers.step_helper import StepHelper
 from src.testproject.rest.messages import DriverCommandReport, CustomTestReport
 from src.testproject.sdk.internal.agent import AgentClient
 from src.testproject.sdk.internal.helpers.redact_helper import RedactHelper
+from src.testproject.sdk.internal.reporter import Reporter
 
 
 class ReportingCommandExecutor:
@@ -300,9 +301,9 @@ class ReportingCommandExecutor:
             always_pass=self.settings.always_pass)
 
         # Handle screenshot condition
-        self.report().step(description=f'Pause for {{{{{milliseconds}}}}} ms',
-                           message=step_message,
-                           inputs={"milliseconds": milliseconds},
-                           passed=result,
-                           screenshot=screenshot)
         screenshot = self.step_helper.take_screenshot(self.settings.screenshot_condition, result)
+        Reporter(self._command_executor).step(description=f'Pause for {{{{{milliseconds}}}}} ms',
+                                              message=step_message,
+                                              inputs={"milliseconds": milliseconds},
+                                              passed=result,
+                                              screenshot=screenshot)

